server/storage/mvcc/key_index.go

```diff
@@ -205,6 +206,18 @@ func (ki *keyIndex) since(lg *zap.Logger, rev int64) []Revision {
 // revision than the given atRev except the largest one (If the largest one is
 // a tombstone, it will not be kept).
 // If a generation becomes empty during compaction, it will be removed.
+//
+// FIXME(fuweid): There is an exception that compact function won't remove
```

Comment:
why is this an exception? `removing the versions with smaller or equal
revision than the given atRev except the largest one` means  revision{1,0} would not be deleted because it is the largest one <= compactRev